### PR TITLE
Expose an endpoint that lists podlings reporting for a given month.

### PR DESCRIPTION
**Pull request by johnament**
Thoughts?

### CODE REVIEW

File: www/roster/main.rb [hunk 1]
```diff
@@ -167,6 +167,14 @@
   _html :podlings
 end
 
+# gives a list of podlings that would report in a given month based on current status
+get '/podlings/reporting/:month.json' do |month|
+  _json ASF::Podling.list
+            .select { |p| p.status == 'current' }
+            .select { |p| (p.reporting.include? month if p.reporting) || (p.monthly.include? month if p.monthly) }
+            .map { |p| { id: p.id, name: p.name, list: p.dev_mail_list } }
```
Comment (rubys) [inline www/roster/main.rb]: I think you want p.display_name

File: www/roster/main.rb [hunk 2]
```diff
@@ -167,6 +167,14 @@
   _html :podlings
 end
 
+# gives a list of podlings that would report in a given month based on current status
+get '/podlings/reporting/:month.json' do |month|
```
Comment (rubys) [inline www/roster/main.rb]: The only real issue here is that this particular tool require authentication; that will make it less easy to use as input to another script.  The code below can be taken pretty much as is (OK, you will need to extract the month from the URL or params for yourself) and made a standalone CGI.

File: www/roster/main.rb (hunk 1 above)
Comment (johnament) [inline www/roster/main.rb]: Yep, that was it.

File: www/roster/main.rb (hunk 2 above)
Comment (johnament) [inline www/roster/main.rb]: Ok, I can see that as an issue, though its weird that I'm not getting prompted.

What do you think about exposing another one of these sinatra apps in a different directory for public info?

Comment (rubys) [inline www/roster/main.rb]: A sinatra application generally require (in addition to your script): a config.ru file, a Gemfile, and a tmp dir.  Such applications are also long running: the process is kept up between requests and only torn down after significant inactivity.  The podling management tool you envision probably should be such a tool.

CGI's scripts that run as a separate process per request.  They also are easier to get started with.  Drop the following file in `whimsy/www/incubator/reporting.cgi` and chmod it to 755:

```
#!/usr/bin/env ruby

$LOAD_PATH.unshift File.realpath(File.expand_path('../../../lib', __FILE__))
require 'wunderbar'
require 'whimsy/asf'

month =  ENV['PATH_INFO'].sub('/', '')

_json do
  ASF::Podling.list
    .select { |p| p.status == 'current' }
    .select { |p| (p.reporting.include? month if p.reporting) ||
                  (p.monthly.include? month if p.monthly) }
    .map { |p| { id: p.id, name: p.display_name, list: p.dev_mail_list } }
end
```

Try it out with http://whimsy.local/incubator/reporting/January

Getting started by writing a big incubator management application may not be the best approach unless you are the type of person that prefers to jump in the deep end.  :-)  Converting the script that you have that sends out reminders to a whimsy/wunderbar application might be an easier first step.

If you point me to the source, I can help; but in the end what you chose to work on and how you chose to implement it is up to you.

Comment (johnament) [inline www/roster/main.rb]: Understood.  And actually, that's why I prefer using sinatra for what this endpoint is doing.  I noticed that the data was loaded once and cached for a period of time, so unlike using cron to write a file, this wouldn't end up being extremely out of date, but shortly out of date if no one is using the ASF::Podling class.

I am the kind of person who jumps deep end first.  I do expect this to go in baby steps, but maybe also refactoring is required (e.g. /roster/ppmc/ shouldn't be protected).  But we can talk about that part later.

**Comment by johnament** [comment]
This is no longer needed.  public_podlings.json now has the data necessary to make this unneeded.

**Pull request closed without merging**